### PR TITLE
fltk: 2-alpha -> 1.3.3

### PR DESCRIPTION
**Pull request by juliendehos**
This PR replace #15614 .
Things done: new branch, nix expression fixed.

### CODE REVIEW

**Comment by vcunat** [comment]
Heh, the diff shown by default isn't really readable due to wrong git heuristics. After reading [upstream description of versions](http://www.fltk.org/articles.php?L825), I think we might even remove `fltk2`; there's no reference in nixpkgs.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -6903,9 +6903,8 @@ in
 
   flite = callPackage ../development/libraries/flite { };
 
-  fltk13 = callPackage ../development/libraries/fltk/fltk13.nix { };
-
-  fltk20 = callPackage ../development/libraries/fltk { };
+  fltk13 = callPackage ../development/libraries/fltk { };
```
Comment (joachifm) [inline pkgs/top-level/all-packages.nix]: Maybe move this to aliases or even remove it outright?

**Comment by juliendehos** [comment]
I removed the expression for fltk2 and wrote 2 aliases (fltk13 and fltk) for 1.3.3.
Travis CI gives an error; I'm trying to solve it...

**Comment by juliendehos** [comment]
Seems ok with the empty "patches".

File: pkgs/development/libraries/fltk/default.nix [hunk 2]
```diff
@@ -39,10 +54,18 @@ composableDerivation.composableDerivation {} rec {
...
   meta = {
-    description = "a C++ cross platform lightweight gui library binding";
+    description = "A C++ cross-platform lightweight GUI library";
     homepage = http://www.fltk.org;
+    platforms = stdenv.lib.platforms.linux;
+    license = stdenv.lib.licenses.gpl2;
   };
+
+  # doesn't build without it. Why?
+  patches = [
```
Comment (jagajaga) [inline pkgs/development/libraries/fltk/default.nix]: That's why :) https://github.com/NixOS/nixpkgs/pull/15622/files#diff-16b346e3b0c0f7128a8404bb4fb4a07cR19
Just rename `postPatch` to something like `preConfigure` or `patchPhase`. If you don't have patches ofc.

Comment (juliendehos) [inline pkgs/development/libraries/fltk/default.nix]: Tried both. Both failed. And now the previous expression doesn't build anymore. Cool...

**Comment by joachifm** [comment]
@juliendehos how are you testing this locally? Don't put too much stock in what travis says.

**Comment by juliendehos** [comment]
Ok. Travis build failed but it's ok locally.

**Comment by vcunat** [comment]
Merged after some additional changes: e5c500475, 34e55827.

**Pull request merged**